gprofiler/main.py

```diff
@@ -609,8 +610,8 @@ def parse_cmd_args() -> configargparse.Namespace:
 
     args = parser.parse_args()
 
-    args.perf_inject = args.nodejs_mode == "perf"
-    args.perf_node_attach = args.nodejs_mode == "attach-maps"
+    args.perf_inject = args.nodejs_mode == "perf" and is_linux()
```

Comment:
having many different branches with `is_linux()` makes it harder to read the function. I prefer if all "windows" changing are done in one site under a single `if`.

So, after `parse_args` you can set `args.flamegraph = False` because that's not supported as we saw.
As for the other perf arguments - do we really need that? I figured just `perf_mode = "disabled"` would be enough? Or maybe even that's not required because the `SystemProfiler` doesn't have any `supported_windows_archs` so it is never selected?

Reply:
Looks better now :)